api/s2n.h

```diff
@@ -66,6 +66,15 @@ typedef enum {
     S2N_EXTENSION_CERTIFICATE_TRANSPARENCY = 18,
 } s2n_tls_extension_type;
 
+typedef enum {
+    S2N_TLS_MAX_FRAG_LEN_EXT_NONE,     /*!< don't use the extension    */
+    S2N_TLS_MAX_FRAG_LEN_512,          /*!< MaxFragmentLength 2^9      */
+    S2N_TLS_MAX_FRAG_LEN_1024,         /*!< MaxFragmentLength 2^10     */
+    S2N_TLS_MAX_FRAG_LEN_2048,         /*!< MaxFragmentLength 2^11     */
+    S2N_TLS_MAX_FRAG_LEN_4096,         /*!< MaxFragmentLength 2^12     */
+    S2N_TLS_MAX_FRAG_LEN_INVALID,      /*!< first invalid value        */
```

Comment:
My intention was to explicitly define valid max fragment codes (values) as defined in RFC6066 with enum values (1~4)
```
 enum{
          2^9(1), 2^10(2), 2^11(3), 2^12(4), (255)
      } MaxFragmentLength;
```
to avoid clients making possible typos by typing in max fragment length directly (ex. 4096) to 

`int s2n_config_set_max_fragment_length(struct s2n_config *config, uint8_t mfl_code);`

Could you explain what you mean by exposing valid parameters will be a "bad" value to the client?